src/backend/utils/misc/guc.c

```diff
@@ -4617,6 +4617,20 @@ static struct config_int ConfigureNamesInt[] =
 		200, 10, MAX_MAX_BACKENDS, NULL, NULL
 	},
 
+	/*
+	 * When HAWQ has one master and one segment deployed together in one physical
+	 * machine, we can not separately set different max connection count. Thus,
+	 * we introduce this guc for segment setting only.
+	 */
+	{
+		{"seg_max_connections", PGC_POSTMASTER, CONN_AUTH_SETTINGS,
+			gettext_noop("Sets the maximum number of concurrent connections in a segment."),
+			NULL
+		},
+		&SegMaxBackends,
+		1600, 80, MAX_MAX_BACKENDS, NULL, NULL
+	},
+
```

Comment:
For default value, according to 1.x  the default is 3 times master value. So we use 200 as master value then 200*3 * 8 we use 4800 as default value.